### PR TITLE
[SELS-TASK] Configure learned words backend

### PR DESCRIPTION
**Pull request by jstephend-sun**
Asana [Task](https://app.asana.com/0/1201478050789792/1201491857751499/f)

expected output:
/api/learned_words is avaible

result: is a list of objects with questions and answers

screenshots:
image

### CODE REVIEW

File: backend/app/Http/Controllers/LearnedWordsController.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+use App\Models\User;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+
+class LearnedWordsController extends Controller {
+    public function index() {
+        $id = Auth::id();
+
+        $user = User::find($id);
```
Comment (ejanton-sun) [inline backend/app/Http/Controllers/LearnedWordsController.php:13]: Laravel provides also to retrieve the authenticated user.
Ref: https://laravel.com/docs/7.x/authentication#retrieving-the-authenticated-user
```suggestion
        $user = Auth::user();
```

File: backend/app/Http/Controllers/LearnedWordsController.php [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        $quiz_items = $quizzes->pluck('quizItems')->collapse();
+        $choices = $quiz_items->pluck('choices')->collapse();
+        $correct_choices = $choices->filter(fn ($choice) => $choice->is_correct);
+
+        $questions = $quiz_items->pluck('question');
+        $answers = $correct_choices->pluck('choice');
+        $array = [];
+        foreach ($questions as $k => $question) {
+            $array[] = ["question" => $question, "answer" => $answers[$k]];
+        }
```
Comment (ejanton-sun) [inline backend/app/Http/Controllers/LearnedWordsController.php]: You can use `map()` in PHP, same in JS. `map()` works in collections.
Ref: https://laravel.com/docs/8.x/collections

**Review by ejanton-sun — CHANGES_REQUESTED** [review]

File: backend/app/Http/Controllers/LearnedWordsController.php (hunk 1 above)
Comment (jstephend-sun) [reply backend/app/Http/Controllers/LearnedWordsController.php:13]: ahhh for this one sir, i need the User Model itself not the Authenticable User

image

File: backend/app/Http/Controllers/LearnedWordsController.php (hunk 2 above)
Comment (jstephend-sun) [reply backend/app/Http/Controllers/LearnedWordsController.php]: done

File: backend/app/Http/Controllers/LearnedWordsController.php (hunk 1 above)
Comment (ejanton-sun) [reply backend/app/Http/Controllers/LearnedWordsController.php:13]: Yet on your line 11, you are retrieving the Authenticated Id.
Auth::user() should return a User instance.

Comment (jstephend-sun) [reply backend/app/Http/Controllers/LearnedWordsController.php:13]: yes sir, Auth::user() returns 

image

Authenticable Class

Whereas i need 
`App\Models\User;`

to access the User model `quiz_logs` function

Comment (ejanton-sun) [reply backend/app/Http/Controllers/LearnedWordsController.php:13]: ahhh the relationship, i see.

Comment (jstephend-sun) [reply backend/app/Http/Controllers/LearnedWordsController.php:13]: > ahhh for this one sir, i need the User Model itself not the Authenticable User
> 
> image

here sir, nag eerror kasi hindi Models\User

**Review by ejanton-sun — APPROVED** [review]

**Pull request merged**